flight/PiOS/STM32F4xx/pios_dacbeep.c

```diff
@@ -0,0 +1,211 @@
+/**
+ ******************************************************************************
+ * @addtogroup PIOS PIOS Core hardware abstraction layer
+ * @{
+  * @addtogroup   PIOS_DACBEEP DAC Beep Code Functions
+ * @brief PIOS interface for DAC Beep implementation
+ * @{
+ *
+ * @file       pios_dacbeep.c
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2015-2016
+ * @brief      Generates audio beeps of desired duration and frequency
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+/* Project Includes */
+#include "pios.h"
+#if defined(PIOS_INCLUDE_DAC_BEEPS)
+
+#if defined(PIOS_INCLUDE_FREERTOS)
+#include "FreeRTOS.h"
+#endif /* defined(PIOS_INCLUDE_FREERTOS) */
+
+#include "pios_dac_common.h"
+
+enum pios_dacbeep_dev_magic {
+	PIOS_DACBEEP_DEV_MAGIC = 0x3A53834A,
```

Comment:
My favorite way to make magic device numbers is to run an md5 on the filename and then grab the first 8 digits. This is (almost) guaranteed to prevent collisions.

I find that making a note of it, like `PIOS_DACBEEP_DEV_MAGIC = 0x7e23b8ab // Magic number generated by md5 of piosdac_beep.c`, keeps my colleagues from guessing where it came from and lets them know how they can generate their own.

This isn't a blocker here, but if you like that approach feel free to use it.